crosshair/condition_parser.py

```diff
@@ -806,10 +809,75 @@ def get_class_invariants(self, cls: type) -> List[ConditionExpr]:
         return []
 
 
+class HypothesisParser(ConcreteConditionParser):
+    @staticmethod
+    def get_hypothesis_test_handle(fn: Callable) -> hypothesis.core.HypothesisHandle:
+        if hasattr(fn, "hypothesis"):
+            return fn.hypothesis
+        else:
+            return None
+
+    def get_fn_conditions(self, ctxfn: FunctionInfo) -> Optional[Conditions]:
+        fn_and_sig = ctxfn.get_callable()
+        if fn_and_sig is None:
+            return None
+        (fn, sig) = fn_and_sig
+        # TODO replace this guard with package-level configuration?
+        if (
+                getattr(fn, "__module__", False)
+                and fn.__module__.startswith("crosshair.")
+                and not fn.__module__.endswith("_test")
+        ):
+            return None
+
+        try:
+            hypothesis_test_handle = HypothesisParser.get_hypothesis_test_handle(fn)
+        except OSError:
+            return None
+        if not hypothesis_test_handle:
+            return None
+
+        inner_test = hypothesis_test_handle.inner_test
+        given_kwargs = hypothesis_test_handle._given_kwargs
```

Comment:
```suggestion
        try:
            inner_test = fn.hypothesis.inner_test
        except AttributeError:
            return None
        given_kwargs = fn.hypothesis._given_kwargs
```

And delete the `get_hypothesis_test_handle` method too since it's now unused.

Reply:
Yep, thats definitely an improvement over what I had.